src/basic/terminal-util.c

```diff
@@ -1370,6 +1366,26 @@ int terminal_set_size_fd(int fd, const char *ident, unsigned rows, unsigned cols
         return 0;
 }
 
+int have_terminfo_file(const char *name) {
+        /* This is a heuristic check if we have the file, using the directory layout used on
+         * current Linux systems. Checks for other layouts can be added later if appropriate. */
+        int r;
+
+        assert(name);
+
+        const char subdir[2] = { name[0] };
+        const char *p = path_join("/usr/share/terminfo", subdir, name);
+        if (!p)
+                return log_oom_debug();
+
+        r = RET_NERRNO(access(p, F_OK));
+        if (r == -ENOENT)
+                return false;
+        if (r == 0)
+                return true;
+        return r;
```

Comment:
as per coding style, the errors should be the indented code paths, the clean code path should be the non-indented one.

hence:

```c
if (access(p, F_OK) < 0)
        return errno == ENOENT ? false : -errno;
return true;
```